src/renderer/components/monaco-editor/monaco-editor.tsx

```diff
@@ -0,0 +1,290 @@
+  language: "yaml",
+  get theme(): MonacoTheme {
+    // theme for monaco-editor defined in `src/renderer/themes/lens-*.json`
+    return ThemeStore.getInstance().activeTheme.monacoTheme;
+  }
+};
+
+@observer
+export class MonacoEditor extends React.Component<MonacoEditorProps> {
+  static defaultProps = defaultEditorProps as object;
+  static viewStates = new WeakMap<Uri, editor.ICodeEditorViewState>();
+
+  static createUri(id: MonacoEditorId): Uri {
+    return Uri.file(`/monaco-editor/${id}`);
+  }
+
+  public staticId = `editor-id#${Math.round(1e7 * Math.random())}`;
+  public dispose = disposer();
+
+  // TODO: investigate how to replace with "common/logger"
+  //  currently leads for stucking UI forever & infinite loop.
+  //  e.g. happens on tab change/create, maybe some other cases too.
+  logger = console;
+
+  @observable.ref containerElem: HTMLElement;
+  @observable.ref editor: editor.IStandaloneCodeEditor;
+  @observable dimensions: { width?: number, height?: number } = {};
+  @observable unmounting = false;
+
+  constructor(props: MonacoEditorProps) {
+    super(props);
+    makeObservable(this);
+  }
+
+  @computed get model(): editor.ITextModel {
+    const uri = MonacoEditor.createUri(this.props.id ?? this.staticId);
+    const model = editor.getModels().find(model => model.uri.toString() == uri.toString());
+
+    if (model) {
+      return model; // already exists
+    }
+
+    const { language, value } = this.props;
+
+    return editor.createModel(value, language, uri);
+  }
+
+  @computed get options(): editor.IStandaloneEditorConstructionOptions {
+    return toJS({
+      ...UserStore.getInstance().editorConfiguration,
+      ...(this.props.options ?? {}),
+    });
+  }
+
+  /**
+   * Monitor editor's dom container element box-size and sync with monaco's dimensions
+   * @private
+   */
+  private bindResizeObserver() {
+    const resizeObserver = new ResizeObserver(entries => {
+      for (const entry of entries) {
+        const { width, height } = entry.contentRect;
+
+        this.setDimensions(width, height);
+      }
+    });
+
+    const containerElem = this.editor.getContainerDomNode();
+
+    resizeObserver.observe(containerElem);
+
+    return () => resizeObserver.unobserve(containerElem);
+  }
+
+  onModelChange = (model: editor.ITextModel, oldModel?: editor.ITextModel) => {
+    this.logger?.info("[MONACO]: model change", { model, oldModel });
+
+    this.saveViewState(oldModel);
+    this.editor.setModel(model);
+    this.restoreViewState(model);
+    this.editor.layout();
+    this.editor.focus(); // keep focus in editor, e.g. when clicking between dock-tabs
+    this.props.onModelChange?.(model, oldModel);
+    this.validateLazy();
+  };
+
+  /**
+   * Save current view-model state in the editor.
+   * This will allow restore cursor position, selected text, etc.
+   * @param {editor.ITextModel} model
+   */
+  saveViewState(model = this.model) {
```

Comment:
I would just remove this default param since there is exactly 2 uses of this method so I think is will cause more confusion then keeping it.

Reply:
Well I don't see any confusion in having current model as default param, but okay..